File: src/uk/co/ruben9922/numbergame/Tile.java
package uk.co.ruben9922.numbergame;

abstract class Tile {
    public enum Colour {
        BLACK, BLUE, ORANGE, RED
    }

    protected Colour colour;

    public Tile(Colour colour) {
        this.colour = colour;
    }

    public Colour getColour() {
        return colour;
    }

    @Override
    public abstract String toString();
}

File: NumberTileGame/Tile.cs
namespace NumberTileGame
{
    internal abstract class Tile
    {
        public Colour Colour { get; set; }

        protected Tile(Colour colour)
        {
            Colour = colour;
        }

        public override string ToString()
        {
            return $"{Colour} Tile";
        }
    }
}